src/pcre2_jit_match_inc.h

```diff
@@ -172,6 +173,17 @@ if (jit_stack != NULL)
 else
   rc = jit_machine_stack_exec(&arguments, convert_executable_func.call_executable_func);
 
+/* XXXXXXXXXX HACK
+Surely there's a better place to put this? I've just stuffed it in here, badly
+I guess it should go into the epilogue of the JIT-generated code somehow,
+conditionally on PCRE2_EXTRA_ALLOW_LOOKAROUND_BSK? */
+if (rc >= 0 &&
+    (ovector[0] < start_offset || ovector[0] > ovector[1]) &&
+    (re->extra_options & PCRE2_EXTRA_ALLOW_LOOKAROUND_BSK) == 0)
+  {
+  rc = PCRE2_ERROR_BAD_BACKSLASH_K;
+  }
```

Comment:
@zherczeg I have done something bad here I think. The code in this file is a tiny wrapper around the JIT, so it feels wrong to implement this check as a post-processing step here.

Reply:
True. This should be done before the pattern returns with a valid match (and PCRE2_HASBSK is set, and PCRE2_EXTRA_ALLOW_LOOKAROUND_BSK is not set).